### PR TITLE
fixes #188: Added Trim Key functionality to the Transit Secret Engine

### LINKED ISSUE #188: Feature Request for Trim Key in Transit Secret Engine
**Describe the feature request or question**
Feature Request for Trim Key API for in the Transit Secret Engine.

**Link to the Vault API Docs that support this feature**
The Vault Docs URL: https://www.vaultproject.io/api/secret/transit#trim-key

**Additional context**
NA

### PR DESCRIPTION
**Pull request by konidev20**
Please review @rajanadar

### CODE REVIEW

File: src/VaultSharp/V1/SecretsEngines/Transit/TransitSecretsEngineProvider.cs [hunk 1]
```diff
@@ -106,5 +106,17 @@ public async Task<Secret<EncryptionResponse>> RewrapAsync(string keyName, Rewrap
                 rewrapRequestOptions)
                     .ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
         }
+
+        public async Task TrimKeyAsync(string keyName, TrimKeyRequestOptions trimKeyRequestOptions, string mountPoint = null)
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/Transit/TransitSecretsEngineProvider.cs:110]: This is integration-testable using program.cs RunTransitSamples method. Feel free to test it locally, if not I'll add the test code before publishing the nuget to verify this.

Comment (konidev20) [reply src/VaultSharp/V1/SecretsEngines/Transit/TransitSecretsEngineProvider.cs:110]: You can go ahead and add the test code.

**Review by konidev20 — COMMENTED** [review]
The requested changes to adding reference in the interface has been taken into consideration and changes have been made.

File: src/VaultSharp/V1/SecretsEngines/Transit/TransitSecretsEngineProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (konidev20) [reply src/VaultSharp/V1/SecretsEngines/Transit/TransitSecretsEngineProvider.cs:110]: Added the function reference into the interface.

**Review by rajanadar — APPROVED** [review]

**Comment by rajanadar** [comment]
@konidev20   New package is published. https://www.nuget.org/packages/VaultSharp/1.6.2.2
Let me know if any issues. There was a bug with json field name that I fixed. Let me know if your trim scenario works.

**Pull request merged**